microcosm_sqlite/stores.py

```diff
@@ -16,11 +16,49 @@
 )
 
 
+def get_session(store):
+    """
+    Return the current session or raise an error.
+
+    """
+    data_set = store.model_class.resolve()
+    session = data_set.session
+    if session is None:
+        raise AttributeError("No session is available in SQLiteContext")
+
+    return session
+
+
+class GetOrCreateSession:
```

Comment:
This is a useful alternative to the above. It basically keeps a long-running session, which should be fine if there are few enough of these and we are read-only, which is a common use case for static DB files.

The benefit here is that we can lazy-init the session and have a model-oriented API (e.g. `Person.search()` vs `graph.person_store.search()`); we've found this idiom to be a better abstraction in self-contained data libraries.